ethereumj-core/src/main/java/org/ethereum/erp/ErpExecutor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.erp;
+
+import org.ethereum.core.Repository;
+import org.ethereum.erp.StateChangeObject.StateChangeAction;
+
+import java.util.Arrays;
+
+import static org.ethereum.util.ByteUtil.EMPTY_BYTE_ARRAY;
+import static org.ethereum.util.ByteUtil.toHexString;
+
+/**
+ * The ERP Executor applies State Change Actions to a Repository.
+ */
+public class ErpExecutor {
+    public static final String WEI_TRANSFER = "weiTransfer";
+    public static final String STORE_CODE = "storeCode";
+
+    public static class ErpExecutionException extends Exception {
+        ErpExecutionException(Throwable cause) {
+            super(cause);
+        }
+    }
+
+    public void applyStateChanges(StateChangeObject sco, Repository repo) throws ErpExecutionException {
+        try {
+            for (StateChangeAction action : sco.actions) {
+                applyStateChangeAction(action, repo);
```

Comment:
Applying code could be a part of `StateChangeAction`. It would be more readable then

Reply:
Ok, I will change to a custom deserializer, which will remove the RawStateChangeObject and will allow me to have separate action classes for each action type.  so the interface will be `action.applyStateChange(repo)`